frontend/src/citizen-frontend/messages/state.tsx

```diff
@@ -59,20 +59,29 @@ const defaultState: MessagePageState = {
 
 export const MessageContext = createContext<MessagePageState>(defaultState)
 
-const markMatchingThreadRead = (
-  threads: MessageThread[],
-  id: UUID
-): MessageThread[] =>
-  threads.map((t) =>
-    t.id === id
-      ? {
-          ...t,
-          messages: t.messages.map((m) => ({
-            ...m,
-            readAt: m.readAt ?? HelsinkiDateTime.now()
-          }))
-        }
-      : t
+export const isRedactedThread = (
+  thread: CitizenMessageThread
+): thread is CitizenMessageThread.Redacted =>
+  thread.type === 'REDACTED_MESSAGE_THREAD'
+export const isRegularThread = (
+  thread: CitizenMessageThread
+): thread is CitizenMessageThread.Regular => thread.type === 'MESSAGE_THREAD'
```

Comment:
note to self: codegen could maybe be extended to generate these automatically for all sealed classes